crates/holochain_persistence_lmdb/src/txn/lmdb.rs

```diff
@@ -0,0 +1,573 @@
+    eav_db: EavLmdbStorage<A>,
+
+    /// Path prefix to generate staging databases
+    staging_path_prefix: PathBuf,
+
+    /// Initial map size of staging databases
+    staging_initial_map_size: Option<usize>,
+
+    /// Environment flags for staging databases.
+    staging_env_flags: Option<EnvironmentFlags>,
+}
+
+/// Name of CAS staging database
+const STAGING_CAS_BUCKET: &str = "staging_cas";
+
+/// Name of EAV staging database
+const STAGING_EAV_BUCKET: &str = "staging_eav";
+
+impl<A: Attribute + DeserializeOwned> CursorProvider<A> for LmdbCursorProvider<A> {
+    type Cursor = LmdbCursor<A>;
+    type CursorRw = LmdbCursor<A>;
+    fn create_cursor_rw(&self) -> PersistenceResult<Self::CursorRw> {
+        let db_names = vec![STAGING_CAS_BUCKET, STAGING_EAV_BUCKET];
+
+        let mut staging_path = self.staging_path_prefix.clone();
+        staging_path.push(format!("{}", Uuid::new_v4()));
+
+        // TODO do we need this if the environment flags are set correctly? That is, it should just
+        // be an in memory only database with no file system handles?
+        fs::create_dir_all(staging_path.as_path())?;
+
+        // This avoids using the singleton rkv manager which caches all environments for all
+        // eternity. As these are randomly pathed staging databases, there is no need for a
+        // singleton to ensure only one environment is created per path.
+        let use_rkv_manager = false;
+
+        let staging_dbs = LmdbInstance::new_all(
+            db_names.as_slice(),
+            staging_path,
+            self.staging_initial_map_size,
+            self.staging_env_flags,
+            use_rkv_manager.into(),
+        );
+
+        let staging_cas_db = LmdbStorage::wrap(
+            staging_dbs
+                .get(&STAGING_CAS_BUCKET.to_string())
+                .unwrap()
+                .clone(),
+        );
+        let staging_eav_db = EavLmdbStorage::wrap(
+            staging_dbs
+                .get(&STAGING_EAV_BUCKET.to_string())
+                .unwrap()
+                .clone(),
+        );
+
+        Ok(LmdbCursor::new(
+            self.cas_db.clone(),
+            self.eav_db.clone(),
+            staging_cas_db,
+            staging_eav_db,
+        ))
+    }
+
+    fn create_cursor(&self) -> PersistenceResult<Self::Cursor> {
+        self.create_cursor_rw()
+    }
+}
+
+pub type LmdbManager<A> =
+    DefaultPersistenceManager<A, LmdbStorage, EavLmdbStorage<A>, LmdbCursorProvider<A>>;
+
+pub fn new_manager<A: Attribute + DeserializeOwned, EP: AsRef<Path> + Clone, SP: AsRef<Path>>(
+    env_path: EP,
+    staging_path_prefix: Option<SP>,
+    initial_map_size: Option<usize>,
+    env_flags: Option<EnvironmentFlags>,
+    staging_initial_map_size: Option<usize>,
+    staging_env_flags: Option<EnvironmentFlags>,
+) -> LmdbManager<A> {
+    let cas_db_name = crate::cas::lmdb::CAS_BUCKET;
+    let eav_db_name = crate::eav::lmdb::EAV_BUCKET;
```

Comment:
It would be nice if the DBs could be nameable for creating multiple DB pairs in the same directory

Reply:
Can consider of future work.